catalog/files/impl/src/test/java/org/projectnessie/catalog/files/adls/TestAdlsProgrammaticOptions.java

```diff
@@ -0,0 +1,122 @@
+                    .endpoint("endpoint1")
+                    .externalEndpoint("externalEndpoint1")
+                    .retryPolicy(AdlsRetryStrategy.FIXED_DELAY)
+                    .maxRetries(4)
+                    .tryTimeout(Duration.ofSeconds(4))
+                    .retryDelay(Duration.ofSeconds(5))
+                    .maxRetryDelay(Duration.ofSeconds(6))
+                    .build())
+            .putFileSystems(
+                "fs2",
+                ImmutableAdlsNamedFileSystemOptions.builder()
+                    .name("my-bucket-2")
+                    .endpoint("endpoint2")
+                    .externalEndpoint("externalEndpoint2")
+                    .retryPolicy(AdlsRetryStrategy.NONE)
+                    .maxRetries(5)
+                    .tryTimeout(Duration.ofSeconds(7))
+                    .retryDelay(Duration.ofSeconds(8))
+                    .maxRetryDelay(Duration.ofSeconds(9))
+                    .build())
+            .build();
+    AdlsOptions expected =
+        ImmutableAdlsProgrammaticOptions.builder()
+            .readBlockSize(1)
+            .writeBlockSize(2)
+            .defaultOptions(
+                ImmutableAdlsNamedFileSystemOptions.builder()
+                    .endpoint("endpoint")
+                    .externalEndpoint("externalEndpoint")
+                    .retryPolicy(AdlsRetryStrategy.EXPONENTIAL_BACKOFF)
+                    .maxRetries(3)
+                    .tryTimeout(Duration.ofSeconds(1))
+                    .retryDelay(Duration.ofSeconds(2))
+                    .maxRetryDelay(Duration.ofSeconds(3))
+                    .build())
+            .putFileSystems(
+                "fs1",
+                ImmutableAdlsNamedFileSystemOptions.builder()
+                    .name("fs1")
+                    .endpoint("endpoint1")
+                    .externalEndpoint("externalEndpoint1")
+                    .retryPolicy(AdlsRetryStrategy.FIXED_DELAY)
+                    .maxRetries(4)
+                    .tryTimeout(Duration.ofSeconds(4))
+                    .retryDelay(Duration.ofSeconds(5))
+                    .maxRetryDelay(Duration.ofSeconds(6))
+                    .build())
+            .putFileSystems(
+                "my-bucket-2",
+                ImmutableAdlsNamedFileSystemOptions.builder()
+                    .name("my-bucket-2")
+                    .endpoint("endpoint2")
+                    .externalEndpoint("externalEndpoint2")
+                    .retryPolicy(AdlsRetryStrategy.NONE)
+                    .maxRetries(5)
+                    .tryTimeout(Duration.ofSeconds(7))
+                    .retryDelay(Duration.ofSeconds(8))
+                    .maxRetryDelay(Duration.ofSeconds(9))
+                    .build())
+            .build();
+    AdlsProgrammaticOptions actual = AdlsProgrammaticOptions.copyOf(input);
+    assertThat(actual).isEqualTo(expected);
+  }
+
+  @Test
+  void normalizeBuckets() {
```

Comment:
Would be good to test the good case as well, and that the bucket options are contained using the `.name()` in S3/GCS/ADLSOptions.